app/services/net_ssh_agent_service.rb

```diff
@@ -0,0 +1,84 @@
+require 'net/ssh'
+require 'fileutils'
+
+class NetSshAgentService
+  attr_reader :pid
+  attr_reader :sock
+
+  def initialize(agent_socket, ssh_private_key)
+    FileUtils.mkdir_p(File.dirname(agent_socket))
+    agent_details = `ssh-agent -a #{agent_socket}`
+    @sock         = agent_details.split("=")[1].split(" ")[0].chop
+    @pid          = agent_details.split("=")[2].split(" ")[0].chop
+    IO.popen({"SSH_AUTH_SOCK" => @sock, "SSH_AGENT_PID" => @pid}, ["ssh-add", "-"], :mode => 'w') do |f|
+      f.puts(ssh_private_key)
+      if $?.to_i != 0
+        raise "Couldn't add key to agent"
+      end
+    end
+  end
+
+  def ssh_exec!(ssh, command, username, ip)
```

Comment:
can you use these methods from the automation states?
if so, we need to remove them (this case from pre_deployment.rb etc.)